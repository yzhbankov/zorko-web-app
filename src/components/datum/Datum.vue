<template>
  <v-layout wrap>
    <v-flex xs3>
      <DatumFieldsPanel/>
    </v-flex>
    <v-flex xs8 style="margin-left: 10px">

    </v-flex>
  </v-layout>
</template>

<script>
  import DatumFieldsPanel from './DatumFieldsPanel.vue'

  export default {
    name: 'Datum',
    components: {
      DatumFieldsPanel
    }
  }
</script>

