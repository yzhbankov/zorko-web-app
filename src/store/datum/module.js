import state from './state'
import * as mutations from './mutations'
import * as actions from './actions'

export default ({namespaced}) => ({
  namespaced,
  state: state(),
  mutations,
  actions
})
