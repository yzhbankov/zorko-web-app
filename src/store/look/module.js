import state from './state'
import * as actions from './actions'
import * as mutations from './mutations'

export default ({namespaced}) => ({
  namespaced,
  state: state(),
  actions,
  mutations
})
