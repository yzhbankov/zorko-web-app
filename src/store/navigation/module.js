import state from './state'
import * as mutations from './mutations'

export default ({namespaced}) => ({
  namespaced,
  state: state(),
  mutations
})
