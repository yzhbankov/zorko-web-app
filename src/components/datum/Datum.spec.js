import {createLocalVue, shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'
import Vuex from 'vuex'
import Datum from './Datum.vue'
import module from '../../store/datum/module'

const localVue = createLocalVue()
const datumModule = module({namespaced: true})

localVue.use(Vuex)

describe('Datum.vue', () => {
  let store
  let actions

  beforeEach(() => {
    actions = {
      loadDatum: jest.fn()
    }
    store = new Vuex.Store({
      state: {},
      modules: {
        datum: {
          ...datumModule,
          actions
        }
      }
    })
  })

  it('initialize', () => {
    const wrapper = shallow(Datum, {store, localVue})

    expect(wrapper.vm).toBeTruthy()
  })

  it('loads datum on create', () => {
    shallow(Datum, {store, localVue})

    expect(actions.loadDatum).toHaveBeenCalled()
  })

  it('has same shallow HTML structure', () => {
    const renderer = createRenderer()
    const wrapper = shallow(Datum, {
      store,
      localVue
    })

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
