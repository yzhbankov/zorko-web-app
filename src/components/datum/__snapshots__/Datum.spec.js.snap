// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datum.vue has same shallow HTML structure 1`] = `
<v-layout wrap="">
    <v-flex xs3="">
        <!---->
    </v-flex>
    <v-flex xs8="" style="margin-left:10px;"></v-flex>
</v-layout>
`;
